client/src/components/overview/GalleryView.jsx

```diff
@@ -124,19 +124,82 @@ function Overview(props) {
         console.log(err);
       })
   }, []);
-  return (
-    // The grid
-    <StyledOverviewGrid>
-      <OverviewCarousel photos={styles[styleIndex].photos}/>
-      <OverViewStars stars={rating}/>
-      <OverViewName name={productInfo.name} category={productInfo.category}/>
-      <OverViewPrice price={styles[styleIndex]}/>
-      <OverViewSelector styles={styles} setStyles={setStyleIndex} styleIndex={styleIndex}/>
-      <OverViewForm styles={styles[styleIndex]}/>
-      <OverviewDescription description={productInfo.description}/>
-      <OverviewFacts facts={productInfo.features}/>
-    </StyledOverviewGrid>
-  )
+
+  const isInitialMount = useRef(true);
+  useEffect(() => {
+    if (isInitialMount.current) {
+     isInitialMount.current = false;
+    } else {
+      var zoomBoxes = document.querySelectorAll('.detail-view');
```

Comment:
Really cool! What is querySelectorAll from?

Reply:
This is for the zoom functionality on the extended view of the pictures. I copy pasted that from a library :D.